### ISSUE TITLE
ssh /home/username/ dir not found

### ISSUE THREAD

**Issue by Konfekt** [msys2/MSYS2-packages#136]
Because sourceforge tickets seem somewhat abandoned, let me paste here the following issue resumed by Ed Morly:

tl;dr: For some reason MSYS2's ssh expansion of "~" doesn't take into account HOME, but the |ls -al| in step 9 finds it with no problems.

STR:
1) Remove/rename old C:\msys64 directory.
2) Set the env variable HOME to "C:\Users\Ed" (replace for your username).
3) Ensure there is no .profile in "C:\Users\Ed" that could interfere.
4) Install msys2-x86_64-20141113.exe
5) Run msys2_shell.bat
6) $ pacman -Sy
7) $ pacman --needed -S bash pacman msys2-runtime
8) Exit the shell.
6) Run msys2_shell.bat
7) $ ssh people.mozilla.org -v

```
    OpenSSH_6.7p1, OpenSSL 1.0.1j 15 Oct 2014
    debug1: Reading configuration data /etc/ssh/ssh_config
    debug1: Connecting to people.mozilla.org [63.245.214.133] port 22.
    debug1: Connection established.
    Could not create directory '/home/Ed/.ssh'.
```

```
...
```

```
8) $ ssh people.mozilla.org -v -F ~/.ssh/config

    OpenSSH_6.7p1, OpenSSL 1.0.1j 15 Oct 2014
    debug1: Reading configuration data /c/Users/Ed/.ssh/config
    debug1: /c/Users/Ed/.ssh/config line 64: Applying options for *
    debug1: Connecting to people.mozilla.org [63.245.214.133] port 22.
    debug1: Connection established.
    debug1: key_load_public: No such file or directory
    debug1: identity file /home/Ed/.ssh/foo_rsa type -1
    ...
```

9) $ ls -al ~/.ssh/config
-rw-r--r-- 1 Ed None 2072 Oct 29 12:11 /c/Users/Ed/.ssh/config

Expected result:
ssh successfully loads the config from ~/.ssh/config (which has an absolute path of /c/Users/Ed/.ssh/config).
When passed the config file using -F, ssh successfully finds the ssh key, which is referenced in the config as "~/.ssh/foo_rsa" (which has absolute path /c/Users/Ed/.ssh/foo_rsa).

Actual result:
ssh doesn't find the config at ~/.ssh/config, and can be seen trying to create a directory at "/home/Ed/.ssh".
When passed the config file using -F, ssh incorrectly expands "~/.ssh/foo_rsa" to "/home/Ed/.ssh/foo_rsa" rather than "/c/Users/Ed/.ssh/foo_rsa".
